theano/tensor/nnet/nnet.py

```diff
@@ -751,7 +751,9 @@ def local_logsoftmax(node):
             isinstance(node.inputs[0].owner.op, Softmax)):
         inVars = node.inputs[0].owner.inputs[0]
         new_op = LogSoftmax()
-        return [new_op(inVars)]
+        new_node = new_op(inVars)
+        copy_stack_trace(node.outputs[0], new_node)
```

Comment:
In that case, we should add both the stack trace of the log (from `node.outputs[0]`) and the one from the softmax (from `node.inputs[0]`), because if logsoftmax raises an error, there are chances the issue was in the original softmax.
You can do `copy_stack_trace([node.inputs[0], node.outputs[0]], new_node).